githubdelivery/src/main/java/io/dockstore/githubdelivery/GithubDeliveryS3Client.java

```diff
@@ -80,29 +93,79 @@ public static void main(String[] args) throws IOException {
             }
         }
 
     }
-    private PushPayload getGitHubDeliveryEventByKey(String key) throws IOException, NoSuchKeyException {
+    private String getObject(String key) throws IOException {
         GetObjectRequest objectRequest = GetObjectRequest
                 .builder()
                 .key(key)
                 .bucket(bucketName)
                 .build();
         ResponseInputStream<GetObjectResponse> object = s3Client.getObject(objectRequest);
-
+        return IOUtils.toString(object, StandardCharsets.UTF_8);
+    }
+    private PushPayload getGitHubPushPayloadByKey(String body, String key) throws IOException, NoSuchKeyException {
         try {
             PushPayload pushPayload;
-            pushPayload = GSON.fromJson(IOUtils.toString(object, StandardCharsets.UTF_8), PushPayload.class);
+            pushPayload = MAPPER.readValue(body, PushPayload.class);
             return pushPayload;
         } catch (JsonSyntaxException e) {
             LOG.error("Could not read github event from key {}", key, e);
             System.exit(1);
         }
         return null;
     }
+    private InstallationRepositoriesPayload getGitHubInstallationRepositoriesPayloadByKey(String body, String key) throws IOException, NoSuchKeyException {
+        try {
+            InstallationRepositoriesPayload installationRepositoriesPayload;
+            installationRepositoriesPayload = MAPPER.readValue(body, InstallationRepositoriesPayload.class);
+            return installationRepositoriesPayload;
+        } catch (JsonSyntaxException e) {
+            LOG.error("Could not read github event from key {}", key, e);
+            System.exit(1);
+        }
+        return null;
+    }
+    private void submitGitHubDeliveryEventsByDate(GithubDeliveryConfig config, String date) {
+
+        ListObjectsV2Request listObjectsV2Request = ListObjectsV2Request
+                .builder()
+                .bucket(bucketName)
+                .prefix(date)
+                .build();
+        ListObjectsV2Iterable listObjectsV2Iterable = s3Client.listObjectsV2Paginator(listObjectsV2Request);
+        for (ListObjectsV2Response response : listObjectsV2Iterable) {
+            response.contents().forEach((S3Object event) -> {
+                submitGitHubDeliveryEventsByKey(config, event.key());
+            });
+        }
+    }
+    private void submitGitHubDeliveryEventsByKey(GithubDeliveryConfig config, String key) {
+        ApiClient apiClient = setupApiClient(config.getDockstoreConfig().serverUrl(), config.getDockstoreConfig().token());
+        WorkflowsApi workflowsApi = new WorkflowsApi(apiClient);
+        String deliveryid = key.split("/")[1]; //since key is in YYYY-MM-DD/deliveryid format
+        try {
+            String body = getObject(key);
+            if (body.contains("\"action\":\"added\"") || body.contains("\"action\":\"removed\"")) {
```

Comment:
These string searches concern me a little bit.  They probably work in perpetuity, but they might break if the json whitespace or quoting changes, or the pattern appears elsewhere for whatever reason.  The latter pattern (`"deleted": true`) is probably more likely to someday show up elsewhere in the event.   If it's easy to parse the event and grab the fields from the parsed representation, that's safer...